pkg/worker/nvidia_smi.go

```diff
@@ -0,0 +1,125 @@
+package worker
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+)
+
+type GpuMemoryUsageStats struct {
+	UsedCapacity  int64
+	TotalCapacity int64
+}
+
+type NvidiaSMIClientInterface interface {
+	AvailableGPUDevices() ([]int, error)
+	GetGpuMemoryUsage(deviceIndex int) (GpuMemoryUsageStats, error)
+}
+
+type NvidiaSMIClient struct{}
```

Comment:
It's really stupid, i mainly add this so that I can easily override functions for testing purposes. In this case (AvailableGPUDevices), the `nvidia-smi` call was causing tests to fail so i wanted a way to easily override this function for those tests.

Reply:
I see. In that case, it might make sense to do the latter which is moving the functions to ContainerCudaManager then embedding an interface into ContainerCudaManager so that it requires AvailableGPUDevices and GetGpuMemoryUsage be implemented. 

This might be better long term in that if we do implement an AMD manager of some sort, it would also require AvailableGPUDevices and GetGpuMemoryUsage be implemented. 

What do you think? Can we give it a try?